flask_appbuilder/views.py

```diff
@@ -570,12 +570,26 @@ def action_post(self):
         pks = request.form.getlist('rowid')
         if self.appbuilder.sm.has_access(name, self.__class__.__name__):
             action = self.actions.get(name)
-            items = [self.datamodel.get(pk) for pk in pks]
+            items = [self.datamodel.get(self._eval_pk(pk)) for pk in pks]
             return action.func(items)
         else:
             flash(as_unicode(FLAMSG_ERR_SEC_ACCESS_DENIED), "danger")
             return redirect('.')
 
+    def _eval_pk(self, pk):
+        """
+        In the case of a model with composite primary keys, `rowid` is a string
+        representation of a list. This method checks for this case and converts
+        it to a list if necessary.
+        """
+        if pk.startswith('[') and pk.endswith(']'):
+            # handle case where composite pk contains datetime column
+            import datetime
+            return eval(pk)
```

Comment:
evals are bad, why is rowid a string rep or a list in the first place? Can we change that instead?

Reply:
@mistercrunch let me see if I can get around this. This was a fairly quick solution using the existing setup, but i agree eval isn't ideal, I'll dig a bit more into it.